File: Data Structures & Algorithms/range-sum-query-2d-immutable/submission-0.cpp
class NumMatrix {
    vector<vector<int>> matrix;
public:
    NumMatrix(vector<vector<int>>& matrix) {
        this->matrix = matrix;
    }
    
    int sumRegion(int row1, int col1, int row2, int col2) {
        int tot = 0;

        for (int i = row1; i<=row2; i ++) {
            for (int j = col1; j <= col2; j ++) {
                tot += matrix[i][j];
            }
        }
        return tot;
    }
};

/**
 * Your NumMatrix object will be instantiated and called as such:
 * NumMatrix* obj = new NumMatrix(matrix);
 * int param_1 = obj->sumRegion(row1,col1,row2,col2);
 */

File: Data Structures & Algorithms/design-hashset/submission-0.py
class MyHashSet:

    def __init__(self):
        self.lim = 10_000
        self.st = [-1] * self.lim

    def add(self, key: int) -> None:
        pos = key%self.lim
        
        for i in range(pos, self.lim):
            if self.st[i] == key: return
            elif self.st[i] == -1: break
        
        self.st[i] = key

    def remove(self, key: int) -> None:
        pos = key%self.lim
        
        for i in range(pos, self.lim):
            if self.st[i] == key: 
               self.st[i] = -1
               break
     
    def contains(self, key: int) -> bool:
        pos = key%self.lim
        for i in range(pos, self.lim):
            if self.st[i] == key: return True
            elif self.st[i] == -1: break
        return False
        


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)